### DISCUSSION TITLE
Question: Timing of deploy with new flag

### DISCUSSION THREAD

**Discussion by elavy-harris** [SagaHealthcareIT/mirthsync#42] [General]
I don't know if this is an issue or not.  (It would be nice for you to add a Discussion section to GitHub for questions.)

The help text for the new "--deploy" flag for mirthSync says, "deploy each included channel immediately after saving the channel to Mirth."

That sounds like a problem, as Code Templates are pushed after Channels, so the deploy might occur before the new Code Template code has been pushed. Is the description wrong (i.e., the deploy happens after the rest of the push is complete) or does the deploy really happen immediately after saving each channel?

FYI, the ReadMe refers to version 3.1.0 instead of 3.0.1.

**Comment by jessesaga** [comment]
Hi, thanks for the heads-up about the readme - I've updated it locally and will fix the version in the next push.

Regarding ordering - Code Templates are pushed before channels so the order shouldn't pose a problem. Are you seeing behavior that would indicate otherwise?

**Reply by elavy-harris** [reply]
No,  Actually, on review of the output from push, I can see that the channels are pushed last, so that's good.

What about dependencies among channels?  Is something done to ensure that dependent channels are deployed after the ones they're dependent on?

**Reply by jessesaga** [reply]
No, I'm not aware of any way to make an automatic determination about channel dependency ordering. 

If you need to control the ordering of channel deployments you could try using the '--restrict-to-path' feature. For instance, you could use the following series of commands to deploy specific channels after pushing everything without deploying....

``` sh
# push everything but don't deploy
mirthsync -s https://server/api -u userx -p passwordx -t targetdir  --deploy push
# push and deploy Channel1
mirthsync -s https://server/api -u userx -p passwordx -t targetdir  --restrict-to-path "Channels/Default Group/Channel1" --deploy push
# push and deploy Channel2
mirthsync -s https://server/api -u userx -p passwordx -t targetdir  --restrict-to-path "Channels/Default Group/Channel2" --deploy push
```

**Reply by elavy-harris** [reply]
It seems that using `/channels/_deploy`, passing in all the appropriate channel IDs, will respect channel deployment dependencies.

Would it be possible to rework the `--deploy` flag so that it uses that API, instead of deploying each channel independently (with `/channels/{channelId}/_deploy`)?
